diff --git a/node_modules/@capacitor/filesystem/ios/Sources/FilesystemPlugin/LegacyFilesystemImplementation.swift b/node_modules/@capacitor/filesystem/ios/Sources/FilesystemPlugin/LegacyFilesystemImplementation.swift
index b9dad01..56fce38 100644
--- a/node_modules/@capacitor/filesystem/ios/Sources/FilesystemPlugin/LegacyFilesystemImplementation.swift
+++ b/node_modules/@capacitor/filesystem/ios/Sources/FilesystemPlugin/LegacyFilesystemImplementation.swift
@@ -131,7 +131,9 @@ import Capacitor
                     let timeElapsed = currentTimestamp - lastEmitTimestamp
 
                     if totalBytesExpectedToWrite > 0 {
-                        if timeElapsed >= 0.1 {
+                        let isTimeToEmit = timeElapsed >= 0.1
+                        let isLastChunk = totalBytesWritten == totalBytesExpectedToWrite
+                        if isTimeToEmit || isLastChunk {
                             emitter(totalBytesWritten, totalBytesExpectedToWrite)
                             lastEmitTimestamp = currentTimestamp
                         }
